feat: add dedicated endpoint for subtask status updates

Emilia-Z opened the pull request.

Emilia-Z committed:
feat: add dedicated endpoint for subtask status updates

nxn-nicole commented on blotztask-api/Modules/Tasks/Commands/SubTasks/UpdateSubtaskStatus.cs:
```diff
@@ -0,0 +1,49 @@
+
+public class UpdateSubtaskStatusCommand
+{
+    [Required]
```
looks like no need to use required here

nxn-nicole commented on blotztask-api/Modules/Tasks/Commands/SubTasks/UpdateSubtaskStatus.cs:
```diff
@@ -0,0 +1,49 @@
+    }
+}
+
+public class SubtaskStatusResultDto
```
look like frontend doesn't need these information? maybe we don't need to return it

nxn-nicole commented on blotztask-api/Modules/Tasks/Commands/SubTasks/UpdateSubtaskStatus.cs:
```diff
@@ -0,0 +1,49 @@
+
+
+
+public class UpdateSubtaskStatusCommand
```
why need an extra command here and doesn't use subtaskId directly?

Emilia-Z committed:
fix: simplify subtask status update endpoint

- Refactored handler to accept subtaskId directly
- Updated handler to return a simple success message instead of full DTO

Emilia-Z committed:
Merge branch 'main' into update-subtask-status

nxn-nicole commented on blotztask-api/Modules/Tasks/Commands/SubTasks/UpdateSubtaskStatus.cs:
```diff
@@ -0,0 +1,38 @@
+
+
+
+public class @int
```
why need to use @int? I feel like you use Handle(int subtaskId, below is fine

Emilia-Z committed:
fix: delete unnecessary int class
